packages/eslint-plugin/tests/rules/no-duplicate-type-constituents.test.ts

```diff
@@ -0,0 +1,647 @@
+    {
+      code: `type T = string[] ${operator} string[];`,
+      output: `type T = string[]  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'string[]',
+            previous: 'string[]',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = string[][] ${operator} string[][];`,
+      output: `type T = string[][]  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'string[][]',
+            previous: 'string[][]',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = [1, 2, 3] ${operator} [1, 2, 3];`,
+      output: `type T = [1, 2, 3]  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: '[1, 2, 3]',
+            previous: '[1, 2, 3]',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = () => string ${operator} string;`,
+      output: `type T = () => string  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'string',
+            previous: 'string',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = () => null ${operator} null;`,
+      output: `type T = () => null  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'null',
+            previous: 'null',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = (arg : string ${operator} string) => void;`,
+      output: `type T = (arg : string  ) => void;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'string',
+            previous: 'string',
+          },
+        },
+      ],
+    },
+    {
+      code: `type T = 'A' ${operator} "A";`,
+      output: `type T = 'A'  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: '"A"',
+            previous: "'A'",
+          },
+        },
+      ],
+    },
+    {
+      code: `type A = "A";
+type T = A ${operator} A;`,
+      output: `type A = "A";
+type T = A  ;`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'A',
+            previous: 'A',
+          },
+        },
+      ],
+    },
+    {
+      code: `type A = "A";
+const a : A ${operator} A = 'A';`,
+      output: `type A = "A";
+const a : A   = 'A';`,
+      errors: [
+        {
+          messageId: 'duplicate',
+          data: {
+            type,
+            duplicated: 'A',
+            previous: 'A',
+          },
+        },
+      ],
+    },
+    {
+      code: noFormat`type A = "A";
+type T = (A) ${operator} (A);`,
+      output: noFormat`type A = "A";
```

Comment:
As of #5379 we don't need to `noFormat` `output`:

```suggestion
      output: `type A = "A";
```

Here and elsewhere.

Also, I don't think we need to `noFormat` this test case's `code` either? Here and elsewhere, I think you can remove a bunch (maybe all) of the `noFormat`s.

Reply:
I didn't know this improvements( #5379 ) were in. Thank you!